### PR TITLE
Add a kickstart test for escrow packets and backup passphrases

### PR DESCRIPTION
**Pull request by dashea**

### CODE REVIEW

File: tests/kickstart_tests/escrow-cert.ks [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+cp -a /tmp/escrow_test $ANA_INSTALL_PATH/root/
+%end
+
+%packages
+volume_key
+%end
+
+%post
+# First, check that the escrow stuff is there
+ls "/root/*-escrow" >/dev/null 2>&1
```
Comment (dashea) [inline tests/kickstart_tests/escrow-cert.ks]: I don't know what's going with these quotes, but I'm pretty sure they're wrong. I think just dropping them, and in the backup-passphrase line below, would work more or less right.

File: tests/kickstart_tests/escrow-cert.ks [hunk 2]
```diff
@@ -0,0 +1,107 @@
+%pre
+# Create an nss database for the escrow certifcate
```
Comment (vpodzime) [inline tests/kickstart_tests/escrow-cert.ks]: typo in *certifcate*

File: tests/kickstart_tests/escrow-cert.ks [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+mkdir -p /tmp/escrow_test/nss
+certutil -d /tmp/escrow_test/nss --empty-password -N
```
Comment (vpodzime) [inline tests/kickstart_tests/escrow-cert.ks]: Does the database need to be created for the escrow to work or is it only needed for the test to verify the results? IOW, do we require people using escrow having this in their kickstarts?

**Comment by vpodzime** [comment]
Looks good to me. Would you mind creating something similar as a test for libblockdev?

File: tests/kickstart_tests/escrow-cert.ks (hunk 3 above)
Comment (dashea) [inline tests/kickstart_tests/escrow-cert.ks]: > Does the database need to be created for the escrow to work or is it only needed for the test to verify the results? IOW, do we require people using escrow having this in their kickstarts?

Normally, the NSS database would be on a separate system, and the public certificate would be available over http. This test needs access to the database in order to decrypt the escrow packets to make sure they work, though.

Comment (vpodzime) [inline tests/kickstart_tests/escrow-cert.ks]: Makes sense to me now. Thanks for the extra info!

**Comment by linux-modder** [comment]
more than willing to help in real world testing of this and libblockdev  ....Assuming a local copy  could be generated / migrated for airgap installs  ?

**Comment by dashea** [comment]
> Assuming a local copy could be generated / migrated for airgap installs

scripts/makeupdates in anaconda has a lot of ways to generate an updates image that you can apply before anaconda is run, or you can generate your own boot.iso: https://www.brianlane.com/creating-the-anaconda-bootiso-with-lorax.html

**Comment by dashea** [comment]
> Would you mind creating something similar as a test for libblockdev

Sure, I'll do that when I get back. It looks like none of the crypto stuff is python 3 compatible (python-cryptography might do what python-nss would have, but I can't find any such thing for libvolume_key bindings), so there might be a lot of shell commands in it.

**Pull request closed without merging**